cmd/prow-job-dispatcher/main_test.go

```diff
@@ -363,3 +363,94 @@ func equalError(t *testing.T, expected, actual error) {
 		t.Errorf("%s: expecting error msg %q, got %q", t.Name(), expected.Error(), actual.Error())
 	}
 }
+
+func TestAddEnabledClusters(t *testing.T) {
+	getProvider := func(cluster string) (api.Cloud, error) {
+		if cluster == "build02" || cluster == "build04" {
+			return "gcp", nil
+		}
+		return "aws", nil
+	}
+	tests := []struct {
+		name    string
+		config  *dispatcher.Config
+		enabled sets.String
+	}{
+		{
+			name:    "Add gcp cluster build04",
+			config:  &dispatcher.Config{},
+			enabled: sets.NewString("build04"),
+		},
+		{
+			name:    "Add multiple clusters, different providers",
+			config:  &dispatcher.Config{},
+			enabled: sets.NewString("build03", "build04", "build05"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			addEnabledClusters(tt.config, tt.enabled, getProvider)
+			for cluster := range tt.enabled {
```

Comment:
I would define an `expect` in `tt` and then compare `tt.expect` and `tt.config` after the function.

Reply:
I left it as I implemented it; I just need check some things in the conf after the function and I feel this way I have a more flexible approach and do not need to recreate some parts of the structure. Tests are not very complicated so I think this trade-off here is OK. Let me know if you really want it to happen.